ntoskrnl/ke/i386/mproc.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Architecture specific source file to hold multiprocessor functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ *              Copyright 2021 Victor Perevertkin <victor.perevertkin@reactos.org>
+ */
+
+/* INCLUDES *****************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+typedef struct _APINFO
+{
+    DECLSPEC_ALIGN(PAGE_SIZE) KIDTENTRY Idt[256];
+    DECLSPEC_ALIGN(PAGE_SIZE) KGDTENTRY Gdt[128];
+    DECLSPEC_ALIGN(16) UINT8 NMIStackData[DOUBLE_FAULT_STACK_SIZE];
+    KIPCR Pcr;
+    ETHREAD Thread;
+    KTSS Tss;
+    KTSS TssDoubleFault;
+    KTSS TssNMI;
+} APINFO, *PAPINFO;
+
+
+typedef struct _AP_SETUP_STACK
+{
+    PVOID ReturnAddr;
```

Comment:
This isn't used anymore, is it?

Reply:
It looks like I still use it as a empty space to put KeLoaderBlock in the right spot. I just push KiSystemStartup into the exact address of this "ReturnAddr" entry of the struct in the apentry.S. Ill look into this more in the morning.

Reply:
This should now be finished. :)